R/appMetadata-quarto.R

```diff
@@ -37,20 +34,34 @@ quartoInspect <- function(appDir = NULL, appPrimaryDoc = NULL) {
     ))
   }
 
-  paths <- c(appDir, file.path(appDir, appPrimaryDoc))
+  json <- suppressWarnings(
+    system2(
+      quarto, c("inspect", shQuote(appDir)),
+      stdout = TRUE, stderr = TRUE
+    )
+  )
+  status <- attr(json, "status")
+
+  if (!is.null(status) && !is.null(appPrimaryDoc)) {
+    json <- suppressWarnings(
+      system2(
+        quarto, c("inspect", shQuote(file.path(appDir, appPrimaryDoc))),
+        stdout = TRUE, stderr = TRUE
+      )
+    )
+    status <- attr(json, "status")
+  }
 
-  for (path in paths) {
-    args <- c("inspect", shQuote(path.expand(path)))
-    inspect <- tryCatch(
-      {
-        json <- suppressWarnings(system2(quarto, args, stdout = TRUE, stderr = TRUE))
-        parsed <- jsonlite::fromJSON(json)
-        return(parsed)
-      },
-      error = function(e) NULL
+  if (!is.null(status)) {
+    cli::cli_abort(
+      c(
+        "Unable to run `quarto inspect` against your content:",
```

Comment:
```suggestion
        "Failed to run `quarto inspect` against your content:",
```

I don't know why but my heart wants "failed" here

Reply:
I'm OK with making your ❤️ happy. Done.